samples/net/https_client/src/main.c

```diff
@@ -95,11 +95,11 @@ int cert_provision(void)
 		}
 	}
 
-	printk("Provisioning certificate\n");
+	printk("Provisioning certificate to the modem\n");
 
 	/*  Provision certificate to the modem */
 	err = modem_key_mgmt_write(TLS_SEC_TAG, MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert,
-				   sizeof(cert) - 1);
```

Comment:
Can you please explain why this change is needed? Has newline handling changed in `modem_key_mgmt_write`, or was our use incorrect?

Reply:
Before this PR, the certificate was written without the last character (`\n` at the end), while in `modem_key_mgmt_cmp()` it is included. So the certificates did not match and it was rewritten. The PR (not removing the last character) aligns with use elsewhere in NCS, while we also add logic to `modem_key_mgmt_cmp()` to make it accept both.